Polyfills/Window/Source/TimeoutDispatcher.cpp

```diff
@@ -0,0 +1,160 @@
+#include "TimeoutDispatcher.h"
+
+#include <cassert>
+
+namespace Babylon::Polyfills::Internal
+{
+    namespace
+    {
+        using TimePoint = std::chrono::time_point<std::chrono::steady_clock, std::chrono::microseconds>;
+
+        TimePoint Now()
+        {
+            return std::chrono::time_point_cast<std::chrono::microseconds, std::chrono::steady_clock>(std::chrono::steady_clock::now());
+        }
+    }
+
+    struct TimeoutDispatcher::Timeout
+    {
+        TimeoutId id;
+
+        // Make this non-shared when JsRuntime::Dispatch supports it.
+        std::shared_ptr<Napi::FunctionReference> function;
+
+        TimePoint time;
+
+        Timeout(TimeoutId id, Napi::Function function, TimePoint time)
+            : id{id}
+            , function{std::make_shared<Napi::FunctionReference>(Napi::Persistent(function))}
+            , time{time}
+        {
+        }
+
+        Timeout(const Timeout&) = delete;
+        Timeout(Timeout&&) = delete;
+    };
+
+    TimeoutDispatcher::TimeoutDispatcher(Babylon::JsRuntime& runtime)
+        : m_runtime(runtime)
+    {
+        m_thread = std::thread{&TimeoutDispatcher::ThreadFunction, this};
+    }
```

Comment:
Note that I moved the initialization of `m_thread` into the body of the constructor to make sure `m_mutex` is initialized before the thread start. While debugging on Ubuntu, the thread kept starting up before `m_mutex` was ready.

Let me know if there is a better way to do this.

Reply:
Variable initialization should be in order of declaration. See [this](https://stackoverflow.com/questions/6308915/member-fields-order-of-construction#:~:text=Member%20variables%20are%20constructed%20in%20the%20order%20in,variables%20appear%20in%20the%20initializer%20list%20is%20irrelevant.).

You just need to move `m_mutex` before `m_thread` in the header. It would have been an uninitialized access otherwise.